src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -240,5 +234,7 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
             return false;
         }
+
+        public SyntaxNode ReturnTypeSyntax => CompilationUnit.Members.First(m => m.Kind() == SyntaxKind.GlobalStatement);
```

Comment:
> Why is this property named ReturnTypeSyntax if it actually returns a global statement?

Because it returns a node that we want to use in scenarios which normally use return type syntax. Since a Simple Program doesn't declare return type explicitly, we return something that is good enough for us. What this node is is an implementation detail and shouldn't really be reflected in the name. The name reflects the purpose of the API not an implementation.

---
In reply to: [403351131](https://github.com/dotnet/roslyn/pull/43038#discussion_r403351131) [](ancestors = 403351131)